### PR TITLE
Create related "research page", fix UI and add @if.

### PR DESCRIPTION
**Pull request by yumeR07**
＊Create research page
HomeController.php
web.php
users/top.blade.php
users/research/show.blade.php
users/research/modals/reservation.blade.php

＊Fix UI
modal.scss
style-users.scss
users/reserved/join_users.blade.php

＊Add @if
users/reserved/show_details.blade.php
　→Added a check for the existence of the participant. (Kenneth's comment on June 22 in Slack.)

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -58,4 +58,18 @@ public function showUser(Meeting $meeting){
...
+        // $all_meetings = $this->meeting->all();
+        // $category = Category::findOrFail($category->id);
+        // $all_meetings = $category->$all_meetings;
+        $all_meetings = $category->meetings->all();
+        $user = Auth::user();
+
+        return view('users.research.show')
+        ->with('all_meetings', $all_meetings)
+        ->with('category', $category)
+        ->with('user', $user);
```
Comment (keanbillyfernandez) [inline app/Http/Controllers/HomeController.php:73]: you can put this in an array

File: resources/views/users/reserved/join_users.blade.php [hunk 2]
```diff
@@ -33,38 +33,25 @@
...
+                                    <div class="avatar mx-auto">
+                                        @if($user->avatar)
+                                        <img src="{{ asset('storage/avatars/'.$user->avatar) }}" alt="" class="avatar-md">
+                                        @else
+                                        <i class="fa-solid fa-circle-user fa-2xl text-secondary text-center"></i>
+                                        @endif
+                                    </div>
+                                    {{-- show user's level icon --}}
+                                    <h3 class="mx-auto text-center" id="username">{{ $user->user_name }}</h3>
+                                    @if($user->comment)
```
Comment (keanbillyfernandez) [inline resources/views/users/reserved/join_users.blade.php:49]: you can also check if the user->comment is null or blank by using == ' ' or Isblank function

File: resources/scss/modal.scss [hunk 3]
```diff
@@ -39,7 +39,7 @@
     width: 60%;
```
Comment (keanbillyfernandez) [inline resources/scss/modal.scss:39]: if this file will use in all modal, you can file name this modal.scss but if in specific modal change the file name into the modal_name of functionality.scss

File: app/Http/Controllers/HomeController.php (hunk 1 above)
Comment (yumeR07) [reply app/Http/Controllers/HomeController.php:73]: I fixed it using compact().

File: resources/views/users/reserved/join_users.blade.php (hunk 2 above)
Comment (yumeR07) [reply resources/views/users/reserved/join_users.blade.php:49]: Thank you for your help.
I could not use the isBlank function, empty function worked correctly. This time I want to use @if, which we used in the previous lesson.

File: resources/scss/modal.scss (hunk 3 above)
Comment (yumeR07) [reply resources/scss/modal.scss:39]: This file is used in all modal, so this file name is fine with modal.scss.

File: app/Http/Controllers/HomeController.php [hunk 4]
```diff
@@ -58,4 +58,11 @@ public function showUser(Meeting $meeting){
         ->with('meeting', $meeting)
         ->with('all_users', $all_users);
     }
+
+    public function showMeeting(Category $category){
+        $all_meetings = $category->meetings->all();
```
Comment (raymonkredo) [inline app/Http/Controllers/HomeController.php:63]: no need for `->all()` actually

Comment (yumeR07) [reply app/Http/Controllers/HomeController.php:63]: It worked without all(). so, I removed all().

File: resources/views/users/research/modals/reservation.blade.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="modal-dialog">
+        <div class="modal-content border border-2 border-warning">
+
+            <div class=“modal-body”>
+                <div class="modal-title text-center fs-2 my-3 fw-bold">Make a Reservation</div>
+                   <p class="modal-text mx-auto mb-0 fs-4">{{ $meeting->category->name }}</p>
+                   <p class="modal-text mx-auto mb-3 fs-4">〜{{ $meeting->title }}〜</p>
+                   <p class="modal-text mx-auto mb-3">Date: {{ $meeting->date }}</p>
+                   <p class="modal-text mx-auto mb-3">Time: {{ $meeting->start_at }}</p>
+                   <p class="modal-text mx-auto mb-3">Level: {{$meeting->level->name }}</p>
```
Comment (krs91tk) [inline resources/views/users/research/modals/reservation.blade.php:11]: add a space before $meeting

File: resources/views/users/research/modals/reservation.blade.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+                </div>
+
+                {{-- button --}}
+                <div class="buttons mt-5 mb-3 text-center">
+                    <button class="button btn-gray" data-bs-dismiss="modal">Cancel</button>
+                    <button type="" class="button btn-orange">Reservation</button>
```
Comment (krs91tk) [inline resources/views/users/research/modals/reservation.blade.php:17]: pls remove type="" if you dont use this

File: resources/views/users/research/show.blade.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+@section('content')
+  <div class="container">
+      <div class="row">
+          <div class="col-8">
+            <div class="category">
+                <h2 class="display-5 mb-3">{{ $category->name }}</h2>
+                  @forelse($all_meetings as $meeting)
+                    <div class="category-myroom mx-auto mb-2">
+                      <div class="category-item">
+                        {{ $meeting->date }}<br>{{ $meeting->start_at}} : 00〜
```
Comment (krs91tk) [inline resources/views/users/research/show.blade.php:17]: pls add a space after start_at

File: resources/views/users/research/show.blade.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+                      </div>
+                      {{-- <div class="category-item">
+                        <img src="image/begginer.png" alt="">
+                      </div> --}}
+                      <div class="category-item">
+                        {{ $meeting->title}}
```
Comment (krs91tk) [inline resources/views/users/research/show.blade.php:23]: pls add a space after title

File: resources/views/users/research/modals/reservation.blade.php (hunk 5 above)
Comment (yumeR07) [reply resources/views/users/research/modals/reservation.blade.php:11]: I added a space before $meeting line11.

File: resources/views/users/research/show.blade.php [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+                      </div>
+                      <div class="category-item">
+                        {{-- <a href="{{ route('', ['meeting' => $meeting->id]) }}" class="text-muted"> --}}
```
Comment (Ditzler01) [inline resources/views/users/research/show.blade.php:26]: Also remove comments if they're not in use.

File: resources/views/users/research/modals/reservation.blade.php (hunk 6 above)
Comment (yumeR07) [reply resources/views/users/research/modals/reservation.blade.php:17]: I remove type="".

File: resources/views/users/research/show.blade.php (hunk 7 above)
Comment (yumeR07) [reply resources/views/users/research/show.blade.php:17]: I added a space after start_at and title.

File: resources/views/users/research/show.blade.php (hunk 8 above)
Comment (yumeR07) [reply resources/views/users/research/show.blade.php:23]: I added a space after title.

File: resources/views/users/research/show.blade.php (hunk 9 above)
Comment (yumeR07) [reply resources/views/users/research/show.blade.php:26]: I use this one. so uncomment it and add route.

**Pull request merged**